lib/functions.inc.php

```diff
@@ -95,6 +98,93 @@ function make_ad_password($password) {
     return $adpassword;
 }
 
+# Returns an array with "scheme" and "salt"
+function user_password_analyzer( $user_password_value ) {
+    $matches = array ();
+    if ( !preg_match('/{(\S+)}(\S+)/', $user_password_value, $matches) ) {
+        throw new InvalidArgumentException('Hashed password does not validate LDAP format');
```

Comment:
How will this error be displayed in the interface? Will it be ignored and just logged?